src/main/java/com/google/sps/servlets/YoutubeServlet.java

```diff
@@ -14,12 +14,74 @@
 
 package com.google.sps.servlets;
 
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.youtube.model.CommentThreadListResponse;
+import com.google.api.services.youtube.YouTube;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
 
 /**
- * Servlet that fetch from Youtube Server.
+ * Servlet that fetches from Youtube Server.
  */
-@WebServlet("/userLogIn")
+@WebServlet("/YouTubeComments")
 public class YoutubeServlet extends HttpServlet {
+  private static final Long COMMENT_LIMIT = 100L;
+  private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+  private static final String URL_PARAMETER = "url";
+  // Parameters required by YouTube API to retrieve the comment threads
+  private static final String SNIPPET_PARAMETERS = "snippet,replies";
+  private static final String ORDER_PARAMETER = "relevance";
+  // TODO: obtain actual API Key
+  private static final String DEVELOPER_KEY = "OUR_API_KEY"; 
+  private static final String APPLICATION_NAME = "SAY";
+
+  /**
+   * Applies parameters to the YouTube API, then extracts comments.
+   * URL is the only true variable; for this application we will always want
+   * order to be relevance and max results to be 100, the API's arbitrary limit
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    try {
+      YouTube youtubeService = getService();
+      String url = request.getParameter(URL_PARAMETER);
+      YouTube.CommentThreads.List commentRequest = youtubeService.commentThreads()
+          .list(SNIPPET_PARAMETERS);
+      CommentThreadListResponse commentResponse = commentRequest.setKey(DEVELOPER_KEY)
+          .setVideoId(url)
+          .setOrder(ORDER_PARAMETER)
+          .setMaxResults(COMMENT_LIMIT)
+          .execute();
```

Comment:
This structuring is a little unclear since the construction of the request and the retrieval of the response are merged/mingled between two lines of code.

Instead I suggest creating a private helper method to abstract the generation of the desired Youtube.CommentThreads.List request. 

This could look something like:
```
Youtube.CommentThreads.List commentRequest = generateYoutubeRequest(url);
CommentThreadListResponse commentResponse = commentRequest.execute();
```

Reply:
While I see how this could be helpful, the way I have it now was emulated from the sample code. For the time being, I think I'm going to keep it as it is for simplicity's sake.

Reply:
Ack. But I think my suggestion (or at least modifying the commentRequest declaration to include the invocation of all of the modifier methods) would make things simpler. 

Since it's unlikely that we're going to need this change much in the future (hopefully) it's worth making this change now unless there's a good reason not to.

Also where did you find the sample code with this form?

Reply:
I agree with Rick here. 
Sample code is meant to be instructive but it's not necessarily how you would do it in a production environment (which is the aim of this project). Rick's suggestion simplifies the code and is good practice. 

Also in the future, "sample code doing it a certain way" is not good reasoning to ignore a suggestion. The code should be justified on its own merits.

Reply:
Alright, since both of you agree on this, I have gone ahead and implemented it.